scrapyd/sqlite.py

```diff
@@ -131,6 +131,12 @@ def clear(self):
         self.conn.execute("delete from %s" % self.table)
         self.conn.commit()
 
+    def get_highest_priority(self):
+        q = "select priority from %s order by priority desc limit 1" \
```

Comment:
but since additional selects are ran for almost all queue method calls
we are not saving cpu cycles or io

Reply:
I introduce SQLite triggers in the 8th commit,
in which project_priority_map would be updated
whenever an INSERT/UPDATE/DELETE occurs.

This is intended to speed up poll() and avoid race conditions,
rather than to save CPU cycles or io.
I think the cost is acceptable.